src/resources.rs

```diff
@@ -2,6 +2,106 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! Structs to manage device resources.
+//!
+//! The high level flow of resource management among the VMM, the device manager, and the device
+//! is as below:
+//! 1) the VMM creates a new device object.
+//! 2) the VMM asks the new device object for its resource requirements.
+//! 3) the VMM allocates resources for the device object according to resource requirements.
```

Comment:
Could you elaborate how the VMM allocate resources? I mean if there is still a kind of vm-allocator crate or not.

Reply:
There is still some sort of resource allocator, which belongs to the vmm.
The VMM first asks each device, "hei, what type of resources do you need?",
then allocates resources from the resource allocator.